drivers/sensor/bmp388/bmp388.c

```diff
@@ -0,0 +1,711 @@
+	 */
+	val->val1 = tmp / 100000;
+	val->val2 = (tmp % 100000) * 10;
+
+	return 0;
+}
+
+static int bmp388_channel_get(const struct device *dev,
+			      enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	switch (chan) {
+	case SENSOR_CHAN_PRESS:
+		bmp388_press_channel_get(dev, val);
+		break;
+
+	case SENSOR_CHAN_DIE_TEMP:
+	case SENSOR_CHAN_AMBIENT_TEMP:
+		bmp388_temp_channel_get(dev, val);
+		break;
+
+	default:
+		LOG_DBG("Channel not supported.");
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+
+static int bmp388_get_calibration_data(const struct device *dev)
+{
+	struct bmp388_data *data = DEV_DATA(dev);
+	struct bmp388_cal_data *cal = &data->cal;
+
+	if (bmp388_read(dev, BMP388_REG_CALIB0, cal, sizeof(*cal)) < 0) {
+		return -EIO;
+	}
+
+	cal->t1 = sys_le16_to_cpu(cal->t1);
+	cal->t2 = sys_le16_to_cpu(cal->t2);
+	cal->p1 = sys_le16_to_cpu(cal->p1);
+	cal->p2 = sys_le16_to_cpu(cal->p2);
+	cal->p5 = sys_le16_to_cpu(cal->p5);
+	cal->p6 = sys_le16_to_cpu(cal->p6);
+	cal->p9 = sys_le16_to_cpu(cal->p9);
+
+	return 0;
+}
+
+#ifdef CONFIG_DEVICE_POWER_MANAGEMENT
+static int bmp388_set_power_state(const struct device *dev,
+				  uint32_t power_state)
+{
+	uint8_t reg_val;
+
+	struct bmp388_data *data = DEV_DATA(dev);
+
+	if (data->device_power_state == power_state) {
+		/* We are already in the desired state. */
+		return 0;
+	}
+
+	if (power_state == DEVICE_PM_ACTIVE_STATE) {
+		reg_val = BMP388_PWR_CTRL_MODE_NORMAL;
+	} else if ((power_state == DEVICE_PM_SUSPEND_STATE) ||
+		   (power_state == DEVICE_PM_OFF_STATE)) {
+		reg_val = BMP388_PWR_CTRL_MODE_SLEEP;
+	} else {
+		return 0;
+	}
+
+	if (bmp388_reg_field_update(dev,
+				    BMP388_REG_PWR_CTRL,
+				    BMP388_PWR_CTRL_MODE_MASK,
+				    reg_val) < 0) {
+		LOG_DBG("Failed to set power mode.");
+		return -EIO;
+	}
+
+	data->device_power_state = power_state;
+
+	return 0;
+}
+
+static uint32_t bmp388_get_power_state(const struct device *dev)
+{
+	struct bmp388_data *ctx = DEV_DATA(dev);
+
+	return ctx->device_power_state;
+}
+
+static int bmp388_device_ctrl(
+	const struct device *dev,
+	uint32_t ctrl_command,
+	void *context,
+	device_pm_cb cb,
+	void *arg)
+{
+	int ret = 0;
+
+	if (ctrl_command == DEVICE_PM_SET_POWER_STATE) {
+		ret = bmp388_set_power_state(dev, *((uint32_t *)context));
+	} else if (ctrl_command == DEVICE_PM_GET_POWER_STATE) {
+		*((uint32_t *)context) = bmp388_get_power_state(dev);
+	}
+
+	if (cb) {
+		cb(dev, ret, context, arg);
+	}
+	return ret;
+}
+#endif /* CONFIG_DEVICE_POWER_MANAGEMENT */
+
+static const struct sensor_driver_api bmp388_api = {
+	.attr_set = bmp388_attr_set,
+#ifdef CONFIG_BMP388_TRIGGER
+	.trigger_set = bmp388_trigger_set,
+#endif
+	.sample_fetch = bmp388_sample_fetch,
+	.channel_get = bmp388_channel_get,
+};
+
+static int bmp388_init(const struct device *dev)
+{
+	struct bmp388_data *bmp388 = DEV_DATA(dev);
+	const struct bmp388_config *cfg = DEV_CFG(dev);
+	uint8_t val = 0U;
+
+	bmp388->bus = device_get_binding(cfg->bus_name);
```

Comment:
I think this should get updated to use DEVICE_DT_GET to get the bus pointer into the configuration structure in ROM at build time, then use device_is_ready() to check that it's initialized properly here, as is being done to existing drivers throughout the tree for 2.6.